src/Cli/dotnet/Parser.cs

```diff
@@ -109,6 +109,24 @@ public static Command GetBuiltInCommand(string commandName)
                 .FirstOrDefault(c => c.Name.Equals(commandName, StringComparison.OrdinalIgnoreCase));
         }
 
+        /// <summary>
+        /// Implements token-per-line response file handling for the CLI. We use this instead of the built-in S.CL handling
+        /// to ensure backwards-compatibility with MSBuild.
+        /// </summary>
+        public static bool TokenPerLine(string tokenToReplace, out IReadOnlyList<string> replacementTokens, out string errorMessage) {
+            var filePath = Path.GetFullPath(tokenToReplace);
+            if (File.Exists(filePath)) {
+                replacementTokens = File.ReadAllLines(filePath).Where(line => !line.StartsWith("#")).Select(line => $"\"{line}\"").ToArray();
```

Comment:
How sure are you that this exactly matches the previous behavior?

Reply:
working through some of the details of that with @jonsequitur right now, will report back

Reply:
wrapping in quotes is no longer necessary - this was an artifacts of some msbuild ;-gobbling that I've now addressed.